source/type/pb_converter.cpp

```diff
@@ -834,27 +834,33 @@ void PbCreator::Filler::fill_pb_object(const nt::StopPointConnection* c, pbnavit
     }
 }
 
-static uint32_t get_st_utc_offset(const nt::StopTime* st, const uint32_t time) {
+static uint32_t get_local_time(const nt::StopTime* st, const uint32_t time) {
     static const auto flag = std::numeric_limits<uint32_t>::max();
-    return time == flag || st->vehicle_journey == nullptr ? 0 : st->vehicle_journey->utc_to_local_offset();
+    uint32_t offset = 0;
+    if (time != flag && st->vehicle_journey != nullptr) {
+        // getting the positive offset to be safe about integer overflow
+        const int32_t day = DateTimeUtils::SECONDS_PER_DAY;
+        offset = (st->vehicle_journey->utc_to_local_offset() % day + day) % day;
```

Comment:
i think a little comment on this line will be very useful one day :) (or just a simple link of stackoverflow)

Reply:
```
If both operands are nonnegative then the remainder is nonnegative; if not, the sign of the remainder is implementation-defined
```
 :joy: